mirrord/cli/src/db_branches.rs

```diff
@@ -64,28 +76,119 @@ async fn status_command(args: &DbBranchesArgs, names: &[String]) -> CliResult<()
 
     let list_params = ListParams::default();
 
-    let branches = api.list(&list_params).await.map_err(|e| {
-        status_progress.failure(Some("failed to list branches"));
+    let mysql_branches = mysql_api.list(&list_params).await.map_err(|e| {
+        status_progress.failure(Some("failed to list MySQL branches"));
         crate::CliError::ListTargetsFailed(mirrord_kube::error::KubeApiError::KubeError(e))
     })?;
 
-    // Filter by names if specified
-    let filtered_branches: Vec<_> = if names.is_empty() {
-        branches.into_iter().collect()
-    } else {
-        branches
-            .into_iter()
-            .filter(|branch| {
-                branch
-                    .metadata
-                    .name
+    let pg_branches = pg_api.list(&list_params).await.map_err(|e| {
+        status_progress.failure(Some("failed to list PostgreSQL branches"));
+        crate::CliError::ListTargetsFailed(mirrord_kube::error::KubeApiError::KubeError(e))
+    })?;
+
+    // Create a unified view with database type information
+    #[derive(Debug)]
+    struct BranchInfo {
+        name: String,
+        db_type: &'static str,
+        phase: String,
+        ttl: u64,
+        database: String,
+        users: String,
+        expire_time: String,
+    }
+
+    let mut all_branches = Vec::new();
+
+    // Process MySQL branches
+    for branch in mysql_branches {
+        if let Some(name) = &branch.metadata.name
+            && (names.is_empty() || names.contains(name))
+        {
+            all_branches.push(BranchInfo {
```

Comment:
nit: maybe cleaner and easier to read if we move these type conversions to `impl From<&Branch> for BranchInfo` from the function.